packages/permissionless/accounts/etherspot/signerToEtherspotSmartAccount.ts

```diff
@@ -0,0 +1,479 @@
+} from "../types"
+import { EtherspotWalletFactoryAbi } from "./abi/EtherspotWalletFactoryAbi"
+import { DUMMY_ECDSA_SIGNATURE, Networks, VALIDATOR_TYPE } from "./constants"
+import { encodeCallData } from "./utils/encodeCallData"
+import { getInitMSAData } from "./utils/getInitMSAData"
+import { getNonceKeyWithEncoding } from "./utils/getNonceKey"
+import { signMessage } from "./utils/signMessage"
+import { signTypedData } from "./utils/signTypedData"
+
+export type EtherspotSmartAccount<
+    entryPoint extends EntryPoint,
+    transport extends Transport = Transport,
+    chain extends Chain | undefined = Chain | undefined
+> = SmartAccount<entryPoint, "etherspotSmartAccount", transport, chain>
+
+/**
+ * The account creation ABI for a modular etherspot smart account
+ */
+const createAccountAbi = [
+    {
+        inputs: [
+            {
+                internalType: "bytes32",
+                name: "salt",
+                type: "bytes32"
+            },
+            {
+                internalType: "bytes",
+                name: "initCode",
+                type: "bytes"
+            }
+        ],
+        name: "createAccount",
+        outputs: [
+            {
+                internalType: "address",
+                name: "",
+                type: "address"
+            }
+        ],
+        stateMutability: "payable",
+        type: "function"
+    }
+] as const
+
+type KERNEL_ADDRESSES = {
+    ecdsaValidatorAddress: Address
+    accountLogicAddress: Address
+    factoryAddress: Address
+    metaFactoryAddress: Address
+    bootstrapAddress: Address
+}
+
+/**
+ * Get default addresses for Etherspot Smart Account based on chainId
+ * @param chainId
+ * @param ecdsaValidatorAddress
+ * @param accountLogicAddress
+ * @param factoryAddress
+ * @param metaFactoryAddress
+ */
+const getDefaultAddresses = (
+    chainId: number,
+    {
+        ecdsaValidatorAddress: _ecdsaValidatorAddress,
+        accountLogicAddress: _accountLogicAddress,
+        factoryAddress: _factoryAddress,
+        metaFactoryAddress: _metaFactoryAddress
+    }: Partial<KERNEL_ADDRESSES>
+): KERNEL_ADDRESSES => {
+    const addresses = Networks[chainId]
+    const ecdsaValidatorAddress =
+        _ecdsaValidatorAddress ?? addresses.multipleOwnerECDSAValidator
+    const accountLogicAddress =
+        _accountLogicAddress ?? addresses.modularEtherspotWallet
+    const factoryAddress =
+        _factoryAddress ?? addresses.modularEtherspotWalletFactory
+    const metaFactoryAddress =
+        _metaFactoryAddress ??
+        addresses?.modularEtherspotWalletFactory ??
+        zeroAddress
+    const bootstrapAddress = addresses.bootstrap ?? zeroAddress
+
+    return {
+        ecdsaValidatorAddress,
+        accountLogicAddress,
+        factoryAddress,
+        metaFactoryAddress,
+        bootstrapAddress
+    }
+}
+
+export const getEcdsaRootIdentifier = (validatorAddress: Address) => {
+    return concatHex([VALIDATOR_TYPE.VALIDATOR, validatorAddress])
```

Comment:
The function name says `root` identifier but we are returning type `validator`. I see there is `VALIDATOR_TYPE.ROOT` also. Can we change the name.

Reply:
Changes done.